File: api/src/Persistence/Entities/WorkLog.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Persistence.Entities;

public class WorkLog : BaseEntity
{
    public int UserId { get; set; }

    public int? WorkplaceId { get; set; }
    public Workplace? Workplace { get; set; }

    public string? Title { get; set; }
    public string? Description { get; set; }

    public DateTime StartedAt { get; set; }
    public DateTime? FinishedAt { get; set; }

    public WorkLogStatus Status { get; set; }

    [NotMapped]
    public TimeSpan Duration
        => FinishedAt == null ? TimeSpan.Zero : (TimeSpan)(FinishedAt - StartedAt);
}

public enum WorkLogStatus
{
    Started,
    Finished
}

File: client/src/app/contracts/worklog.ts
export interface WorkLogDetailModel {
  id: number;
  title: string | null;
  description: string | null;
  workplaceId: number | null;
  workplaceName: string | null;
  startedAt: string;
  finishedAt: string | null;
  createdAt: string;
  lastEditedAt: string | null;
  duration: string;
  status: WorkLogStatus;
}

export interface WorkLogCreateModel {
  title: string;
  description: string | null;
  startedAt: string;
  finishedAt: string;
  workplaceId: number | null;
}

export interface WorkLogUpdateModel {
  id: number;
  title: string;
  description: string | null;
  startedAt: string;
  finishedAt: string;
  workplaceId: number | null;
}

export interface WorkLogStartModel {
  title: string;
}

export enum WorkLogStatus {
  Started,
  Finished,
}
